Content.Server/_Shitmed/Body/Organ/HeartSystem.cs

```diff
@@ -29,15 +38,52 @@ private void HandleRemoval(EntityUid uid, HeartComponent _, ref OrganRemovedFrom
 
         // TODO: Add some form of very violent bleeding effect.
         EnsureComp<DelayedDeathComponent>(args.OldBody);
+        _alert.ShowAlert(args.OldBody, _faultyHeartAlertId); // Omu
     }
 
     private void HandleAddition(EntityUid uid, HeartComponent _, ref OrganAddedToBodyEvent args)
     {
         if (TerminatingOrDeleted(uid) || TerminatingOrDeleted(args.Body))
             return;
 
-        if (_bodySystem.TryGetBodyOrganEntityComps<BrainComponent>(args.Body, out var _))
+        // Omu Edit Start - This a long one
+        // Lets make sure the brain is present and the heart inserted is functioning
+        // TODO: If ever we have something with multiple brains and braindamage is implemented this needs to be changed.
+        if (_bodySystem.TryGetBodyOrganEntityComps<BrainComponent>(args.Body, out var _)
+            && TryComp<OrganComponent>(args.Part, out var organ)
+            && organ.Enabled)
+        {
             RemComp<DelayedDeathComponent>(args.Body);
```

Comment:
Why are we removing? `RemComp<DelayedDeathComponent>` this can cause issues with some systems.